### ISSUE TITLE
Wrapped Text is forced to be left aligned

### ISSUE THREAD

**Issue by Emersont1** [mobius3/kiwi#47]
Code:
```cpp
auto title_l = KW_CreateLabel(gui, frame,"Text Wrapping in Scrollbox\n(Horizontal Bar Disabled)", &r);
//KW_LabelEnableWrap(title_l);
KW_SetLabelAlignment(title_l, KW_LABEL_ALIGN_RIGHT, 0, KW_LABEL_ALIGN_MIDDLE, 0);
```
No wrapping, but centered

![image](https://user-images.githubusercontent.com/23140112/81483081-5ed18c00-9233-11ea-9dd3-d3b534140efc.png)

when I uncomment the `LabelEnableWrap(` line:

![image](https://user-images.githubusercontent.com/23140112/81483097-790b6a00-9233-11ea-86bb-371582620568.png)

The same occurs when it is set to right aligned

I'm logging this, if i have time after the checkboxes and radio buttons I'm working on, I'll work on a drop in replacement for the SDL_TTF rendering function, but this may take time

**Comment by mobius3** [comment]
What are you planning on replacing SDL_TTF with? Replacing SDL_ttf is in my plans as well, right now I'm working on a font cooking library, it's just not public yet, that I plan to use for KiWi 1.0 and other projects.

**Comment by Emersont1** [comment]
I'm thinking using something like freetype, and iirc the ICU library has functionality for wrapping text

**Comment by mobius3** [comment]
Yeah, this new library of mine is a wrapper around stb_truetype (which wasn't a lot easy to work with, to be honest). It already correctly produces  the glyph atlas but I'm still dealing with some kerning issues. I hope to publish it by this week but replacing KiWi's text rendering with it (or any other library) might be a little daunting. I'm not hoping to do it for the master branch but rather to the development branch (which is going to be 1.0), since there will be more liberty to redefine the API as needed.

As for the issue at hand, I'm not sure of what to do - maybe fixing SDL's wrapping is a better idea, for the time being. Until then or until kiwi text rendering is replace, I think documenting at as a library caveat is proper, unless you absolutely need center/aligned wrapped text.

**Comment by Emersont1** [comment]
at present, i don't, but there probably will be issues when it comes to  LTR languages (like Arabic, hebrew) so it ought to be fixed eventually. I think it just needs documenting for now
